### ISSUE TITLE
Import of MR images compressed with JPEG2000 not working

### ISSUE THREAD

**Issue by fordanic** [slicebox/slicebox#91]
Tried to import a set of MR images compressed with JPEG2000 but received the following error message:
2017-01-31 10:57:26,675 INFO  (akka://slicebox) - System: The presentation context [SOPClassUID = 1.2.840.10008.5.1.4.1.1.4, TransferSyntaxUID = 1.2.840.10008.1.2.4.90] is not supported.

**Comment by KarlSjostrand** [comment]
Slicebox does currently not support JPEG2000 transfer syntaxes due to software license incompatibilities between slicebox (Apache 2) and the Java-based JPEG2000 ImageIO plugins that we are aware of. It would be a great feature, though. Any suggestions on how to move forward are most welcome.

**Comment by fordanic** [comment]
Would dcm4che or PixelMed™ Java DICOM Toolkit be possible alternatives?

**Comment by KarlSjostrand** [comment]
Yes, Slicebox uses dcm4che3 internally which supports JPEG2000 through ImageIO plugins. The challenge is to find up to date plugins for JPEG2000. (See e.g. https://github.com/haraldk/TwelveMonkeys). Here is a thread discussing this for dcm4che: https://groups.google.com/forum/#!topic/dcm4che/kxoryXjsidw.

One way forward to consider is to support more transfer syntaxes but just not supporting viewing of such images (which is the only situation in which Slicebox reads and interprets pixel data).

I'm leaving this issue open in case there is more input and/or suggestions.

**Comment by fordanic** [comment]
Maybe a possibility is to use GDCM and their JAVA support to simply decompress JPEG 2000 compressed DICOM files?

**Comment by KarlSjostrand** [comment]
As suggested by @ghisvail, try adding support through https://github.com/jai-imageio/jai-imageio-jpeg2000. If this works, we can provide detailed instructions on how to add this library to a deployed instance. This would work around any licensing issues that would arise from including the library in our repository and binaries.

**Comment by ghisvail** [comment]
The idea being that the regular release of `slicebox` **should not** bundle JP2 support to avoid licensing issues, but the latter may be enabled retrospectively via an optional configuration / install step.

**Comment by ghisvail** [comment]
Unless a patent-free JP2 decoder for JAVA is found in the meantime.

**Comment by ghisvail** [comment]
JP2 can be decoded by OpenJPEG using [imageio-jnr](https://github.com/dbmdz/imageio-jnr).

**Comment by KarlSjostrand** [comment]
@ghisvail Did you find the time to try this yet? Curious if it works.
